src/test/java/org/folio/service/RequestEventHandlerTest.java

```diff
@@ -1,40 +1,197 @@
+import org.folio.spring.service.SystemUserScopedExecutionService;
+import org.folio.support.KafkaEvent;
+import org.folio.support.KafkaEvent.EventType;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.mock.mockito.MockBean;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
-class RequestEventHandlerTest extends BaseIT {
-  private static final String REQUEST_UPDATE_EVENT_SAMPLE =
-    getMockDataAsString("mockdata/kafka/secondary_request_update_event.json");
+@ExtendWith(MockitoExtension.class)
+class RequestEventHandlerTest {
+  private static final UUID PRIMARY_REQUEST_ID = randomUUID();
+  private static final UUID SECONDARY_REQUEST_ID = PRIMARY_REQUEST_ID;
+  private static final UUID INTERMEDIATE_REQUEST_ID = PRIMARY_REQUEST_ID;
+  private static final String PRIMARY_REQUEST_TENANT_ID = "primary_tenant";
+  private static final String SECONDARY_REQUEST_TENANT_ID = "secondary_tenant";
+  private static final String INTERMEDIATE_REQUEST_TENANT_ID = "intermediate_tenant";
 
-  @MockBean
+  @Mock
   private DcbService dcbService;
-  @MockBean
+  @Mock
   RequestService requestService;
-  @MockBean
+  @Mock
   private EcsTlrRepository ecsTlrRepository;
+  @Mock
+  private ServicePointService servicePointService;
+  @Mock
+  private SystemUserScopedExecutionService executionService;
+  @Mock
+  private CloningService<ServicePoint> servicePointCloningService;
 
-  @Autowired
-  private KafkaEventListener eventListener;
+  @InjectMocks
+  private RequestEventHandler handler;
+
+  @Captor
+  ArgumentCaptor<Request> requestCaptor;
+
+  @Test
+  void secondaryRequestInterimPickupServicePointShouldNotBeUpdatedUponPrimaryRequestUpdate() {
+    EcsTlrEntity ecsTlr = buildEcsTlr();
+
+    Request primaryRequest = new Request()
+      .id(PRIMARY_REQUEST_ID.toString())
+      .ecsRequestPhase(PRIMARY)
+      .status(Request.StatusEnum.OPEN_IN_TRANSIT)
+      .fulfillmentPreference(DELIVERY)
+      .deliveryAddressTypeId(randomId())
+      .requestExpirationDate(new Date());
+
+    Request secondaryRequest = new Request()
+      .id(SECONDARY_REQUEST_ID.toString())
+      .ecsRequestPhase(SECONDARY)
+      .fulfillmentPreference(HOLD_SHELF)
+      .pickupServicePointId(INTERIM_SERVICE_POINT_ID);
+
+    when(ecsTlrRepository.findBySecondaryRequestId(PRIMARY_REQUEST_ID))
+      .thenReturn(Optional.of(ecsTlr));
+
+    when(requestService.getRequestFromStorage(SECONDARY_REQUEST_ID.toString(), SECONDARY_REQUEST_TENANT_ID))
+      .thenReturn(secondaryRequest);
+
+    KafkaEvent<Request> event = buildRequestUpdateEvent(primaryRequest, primaryRequest,
+      PRIMARY_REQUEST_TENANT_ID);
+
+    handler.handle(event);
+
+    verifyNoInteractions(dcbService);
```

Comment:
What if we just insert blank lines between meaningful blocks of code instead of after every line?